src/orquestra/quantum/circuits/_circuit.py

```diff
@@ -147,6 +147,19 @@ def __repr__(self):
             f"n_qubits={self.n_qubits})"
         )
 
+    def inverse(self) -> "Circuit":
+        """Create a circuit that, when applied to the current circuit, will
+        produce the identity circuit.): """
+        try:
+            return type(self)(
+                operations=[op.dagger() for op in reversed(self.operations)],
+                n_qubits=self.n_qubits,
+            )
+        except AttributeError:
+            raise NotImplementedError(
+                "Inverse is not implemented for this circuit type."
+            )
```

Comment:
Can we be a little more specific with the error so that the user can figure out why their circuit could not be inverted? If not it's ok Just a suggestion to help users.